worker/jobTypes/githubJob.js

```diff
@@ -41,13 +41,97 @@ class GitHubJobClass {
         return false;
     }
 
+    async constructPrefix(isProdDeployJob){        
+      try{
+        //download published branches file to retrieve prefix and check if repo is versioned 
+        const repoObject = {
+          repoOwner: this.currentJob.payload.repoOwner, repoName: this.currentJob.payload.repoName,
+        };
+        const repoContent = await workerUtils.getRepoPublishedBranches(repoObject)
+        const server_user = await this.getUser()
+        let pathPrefix; 
+        if(isProdDeployJob){
+          //versioned repo
+          if(repoContent && repoContent.content.version.active.length > 1){
+            repoContent.content.prefix
+            pathPrefix = `${repoContent.content.prefix}/${this.currentJob.payload.branchName}`; 
+          }
+        }
+        // server staging commit jobs
+        else if(this.currentJob.payload.patch && this.currentJob.payload.patchType === 'commit'){
+          pathPrefix = `${repoContent.content.prefix}/${this.currentJob.user}/${this.currentJob.payload.localBranchName}/${server_user}/master`; 
+        }
+        //mut only expects prefix or prefix/version for versioned repos, have to remove user from staging prefix
+        if(typeof pathPrefix !== 'undefined' && pathPrefix !== null){
+          const mutPrefix = pathPrefix.split(`/${server_user}`)[0];
+          return[pathPrefix, mutPrefix, server_user]
+        }else{
+          return [null, null, server_user]
+        }
+      }catch(error){
+        console.log(error)
+        throw error
+      }
+        
+    }
+
+    async writeEnvFile(isProdDeployJob){
+        try {
+          const [pathPrefix, mutPrefix, server_user] = await this.constructPrefix(isProdDeployJob)
+          let envVars;
+
+          if(pathPrefix !== null){
+            envVars = 
+            `GATSBY_PARSER_USER=${server_user}
```

Comment:
since both conditional clauses write the same thing for GATSBY_PARSER_USER and GATSBY_PARSER_BRANCH, this can be condensed.